### ISSUE TITLE
Suspected Memory Leak

### ISSUE THREAD

**Issue by irwineffect** [PalouseRobosub/robosub-2018#248]
I logged into my computer today and discovered that I was completely out of RAM and swap after running the simulator for a few days. Not sure which node(s) is causing the leak, more investigation needs to be done. I first suspect the vision processing, I noticed a lot of ram was freed when I killed the buoy nodes.

Running nodes:
- robosub_simulator/gazebo.launch
- robosub/vision.launch
- robosub/jirwin_gate_task.py

**Comment by Sindorman** [comment]
Did you run simulator for days without stopping?

**Comment by skallaher** [comment]
@Sindorman, that seems to be the case. Have you seen any symptoms of this problem?

**Comment by Sindorman** [comment]
@skallaher I tried to recreate same. I run simulator making sub face buoys, I also was running rqt plagin of left camera(if that helps). I took a screenshot of some of htop information, and left it run over night. I did notice change of total memory usage from 2.56 GB to 2.86-3.06 Gb, however the percentage of memory usage stayed the same on simulator. But the number of threads were 20 more then when I left it.

P.S. I will try to see if I will be able to pin-point why, using specific tools, because my test was just general overall if the memory increases over time.
